libs/server/kiln_server/document_api.py

```
@@ -0,0 +1,695 @@
+logger = logging.getLogger(__name__)
+
+# keep track of locks by extractor config ID to prevent concurrent runs of the same extractor config
+# maps from extractor config ID -> lock
+run_extractor_locks: Dict[str, asyncio.Lock] = defaultdict(asyncio.Lock)
+
+
+def open_folder(path: str | Path) -> None:
+    if sys.platform.startswith("darwin"):
+        subprocess.run(["open", path], check=True)
+    elif sys.platform.startswith("win"):
+        os.startfile(path)  # type: ignore[attr-defined]
```

Pylint (3.3.7)

[error] Module 'os' has no 'startfile' member

(E1101)